doc/source/ray-observability/user-guides/configure-logging.md

```diff
@@ -0,0 +1,382 @@
+RAY_ROTATION_BACKUP_COUNT=1; ray start --head # Start a ray instance with backupCount 1.
+```
+
+## Using Ray's logger
+When ``import ray`` is executed, Ray's logger is initialized, generating a default configuration given in ``python/ray/_private/log.py``. The default logging level is ``logging.INFO``.
+
+All Ray loggers are automatically configured in ``ray._private.ray_logging``. To modify the Ray logger:
+
+```python
+import logging
+
+logger = logging.getLogger("ray")
+logger # Modify the Ray logging config
+```
+Similarly, to modify the logging configuration for Ray AIR or other libraries, specify the appropriate logger name:
+
+```python
+import logging
+
+# First, get the handle for the logger you want to modify
+ray_air_logger = logging.getLogger("ray.air")
+ray_data_logger = logging.getLogger("ray.data")
+ray_tune_logger = logging.getLogger("ray.tune")
+ray_rllib_logger = logging.getLogger("ray.rllib")
+ray_train_logger = logging.getLogger("ray.train")
+ray_serve_logger = logging.getLogger("ray.serve")
+ray_workflow_logger = logging.getLogger("ray.workflow")
+
+# Modify the ray.data logging level
+ray_data_logger.setLevel(logging.WARNING)
+
+# Other loggers can be modified similarly.
+# Here's how to add an aditional file handler for Ray Tune:
+ray_tune_logger.addHandler(logging.FileHandler("extra_ray_tune_log.log"))
+```
+(structured-logging)=
+## Structured logging
+Implement structured logging to enable downstream users and applications to consume the logs efficiently.
+
+### System logs
+Ray’s system or component logs are mostly structured following this format. <br />
+(placeholder for the format)
+
+Example <br />
+(placeholder for the example)
+
+:::{note}
+Some system component logs are not structured as suggested above as of 2.5. The migration of system logs to structured logs is ongoing.
+:::
+
+### Application logs
+A Ray applications include both driver and worker processes. For Python applications, use Python loggers to format and structure your logs. 
+As a result, Python loggers need to be set up for both driver and worker processes.
+
+::::{tab-set}
+
+:::{tab-item} Ray Core
+Set up the Python logger for driver and worker processes separately:
+1. Set up the logger for the driver process after importing `ray`.
+2. Use `worker_process_setup_hook` to configure the Python logger for all worker processes.
+
+![Set up python loggers](../images/setup-logger-application.png)
+
+To control the logger for specific Actors or Tasks, view 
+
+:::
+
+:::{tab-item} Ray AIR or other libraries
+If you are using Ray AIR or any of the Ray libraries, follow the instructions provided in the documentation for the library.
+:::
+
+::::
+
+### Add metadata to structured logs
+If you need additional metadata to make logs more structured, fetch the metadata of Jobs, Tasks or Actors with Ray’s [`runtime_context APIs`](#). 
+::::{tab-set}
+
+:::{tab-item} Ray Job
+...
```

Comment:
Need @rkooo567 's input

Reply:
how should we do this? should I write a comment here?

Reply:
job : `get_job_id()` 
actor: `get_actor_id()`
task: `get_task_id()`
worker: `get_worker_id()`

Note that the job submission id is not supported yet.

Reply:
> :::{tab-item} AIR or library-specific entities

Let's remove it for now. I will update this part as a part of my logging PRD work

Reply:
Done